webapp/app.py

```diff
@@ -1161,3 +1163,26 @@ def render_supermicro_blogs():
 
 
 app.add_url_rule("/supermicro", view_func=render_supermicro_blogs)
+
+def render_form(templatePath, formId, isModal, modalId):
+    @wraps(render_form)
+    def wrapper_func():
+        resultForm = {}
+        with app.app_context() and app.test_request_context():
+            filename = os.path.join(app.static_folder, 'files', 'forms-data.json')
+            with open(filename) as form_file:
+                data = json.load(form_file)
+                for form in data["forms"]:
+                    if form["formData"]["formId"] == formId:
+                        resultForm = form
```

Comment:
```suggestion
              resultForm = next(
                    (
                        form
                        for form in data["forms"]
                        if form["formData"]["formId"] == formId
                    ),
                    None,
                )
```
Looping through all the forms will get expensive as the number of forms grows. You can use the [next function](https://www.w3schools.com/python/ref_func_next.asp)  to only loop through the minimum amount of forms.

Reply:
Alternatively, you can create a hash map within the JSON and instead of looping through, use a unique identifier to directly fetch the relevant  form data. See my comment here

Reply:
The above comment will improve performance but require a little refactoring of the code here